source/common/upstream/edf_scheduler.h

```diff
@@ -67,6 +70,21 @@ template <class C> class EdfScheduler {
     ASSERT(queue_.top().deadline_ >= current_time_);
   }
 
+  /**
+   * Insert entry into queue with a given weight only in refreshing stage. The deadline will be
+   *  multiples of (1 / weight) that just larger than initial current time
+   * @param weight floating point weight.
+   * @param entry shared pointer to entry, only a weak reference will be retained.
```

Comment:
My understanding from glancing at the code is that you're essentially rotating the priority queue by ensuring the new deadline comes after the most distant one already in the heap and this would push the entry to the "back of the line" so to speak. Is this correct? If so, can you make this comment a bit more verbose to explain it and consider rewording the function to make this more clear?